internal/consumer/repo.go

```diff
@@ -0,0 +1,59 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+type DbRepository interface {
+	SaveItem(item Item) error
+}
+
+type MongoRepository struct {
+	Client *mongo.Client
+	Ctx    context.Context
+}
+
+func (mr MongoRepository) SaveItem(item Item) error {
+	database := mr.Client.Database("hacker-news")
+	itemsCollection := database.Collection("items")
+
+	_, err := itemsCollection.InsertOne(mr.Ctx, item)
```

Comment:
How are you planning on handling any potential duplication? Could consider Upsert?
Are you using a partition ID?

Reply:
good shout i have changed to Upsert. What is a partition ID in this context?